auth/simple_token.go

```diff
@@ -52,6 +54,21 @@ func (as *authStore) assignSimpleTokenToUser(username, token string) {
 	}
 
 	as.simpleTokens[token] = username
+	deleteTokenFunc := func() {
+		as.simpleTokensMu.Lock()
+		_, ok := as.simpleTokenLock[token]
+		if ok {
+			delete(as.simpleTokenLock, token)
+			as.simpleTokensMu.Unlock()
+			return
+		}
+		delete(as.simpleTokens, token)
+		delete(as.simpleTokenInf, token)
+		as.simpleTokensMu.Unlock()
+		plog.Errorf("deleted the token %s for user %s", token, username)
+	}
+	timer := time.AfterFunc(SimpleTokenTTL, deleteTokenFunc)
```

Comment:
The timer approach does eliminate the routine problem and is very performant. however, it is too easy to mess up: see https://golang.org/pkg/time/#Timer.Reset. 

Also this approach spreads the timer management logical all over the place. I still prefer a go routine that in our control to scan the deadline periodically. I feel it would be cleaner.